src/main/java/com/vinylteam/vinyl/web/controller/CatalogController.java

```diff
@@ -20,12 +27,36 @@
 public class CatalogController {
 
     private final UniqueVinylService uniqueVinylService;
+    private final DefaultOneVinylOffersService oneVinylOffersService;
+
     private final UniqueVinylMapper uniqueVinylMapper;
+    private final OneVinylOfferMapper oneVinylOfferMapper;
 
     @GetMapping
     public List<UniqueVinylDto> getCatalogPage() {
         List<UniqueVinyl> uniqueVinyls = uniqueVinylService.findRandom(50);
         return uniqueVinylMapper.uniqueVinylsToUniqueVinylDtoList(uniqueVinyls);
     }
 
+    @GetMapping("/{id}")
+    public OneVinylPageDto getOneVinylOfferPage(@SessionAttribute(value = "user", required = false) User user,
+                                                @PathVariable("id") String id) throws NotFoundException {
+        UniqueVinyl uniqueVinyl = uniqueVinylService.findById(id);
+        HashMap<String, List> offersAndShopsMap = oneVinylOffersService.getSortedInStockOffersAndShops(id);
```

Comment:
First of all we need to make List generic one and second of all use Map instead of HashMap for link declaration